### PR TITLE
Revamp partition sketching

### PR DESCRIPTION
**Pull request by mavam**
This PR introduces a revamped partition sketching infrastructure.

Details follow soon.

### :memo: Checklist

- [ ] All user-facing changes have changelog entries.
- [ ] The changes are reflected on [docs.tenzir.com/vast](https://docs.tenzir.com/vast), if necessary.
- [ ] The PR description contains instructions for the reviewer, if necessary.

### :dart: Review Instructions

Not yet.

### CODE REVIEW

File: libvast/vast/sketch/sketch.hpp [hunk 1]
```diff
@@ -0,0 +1,37 @@
...
+  /// Checks whether the sketch fulfils a given predicate.
+  /// @param pred The predicate to check.
+  /// @returns A boolean that indicates whether *x* is in the sketch, or
+  /// std::nullopt if the query cannot be answered.
+  std::optional<bool> lookup(relational_operator op, const data& x) const noexcept;
+
+  friend size_t mem_usage(const sketch& x);
+
+private:
+  chunk_ptr flatbuffer_;
```
Comment (mavam) [inline libvast/vast/sketch/sketch.hpp:34]: @dominiklohmann would this be a use case for `flatbuffer<T>`?

Comment (dominiklohmann) [reply libvast/vast/sketch/sketch.hpp:35]: Yes, and similarly so the constructor, which should already receive a `flatbuffer<T>`.

Comment (mavam) [reply libvast/vast/sketch/sketch.hpp]: Does 78a5c02 look good to you in this regard?

Comment (dominiklohmann) [reply libvast/vast/sketch/sketch.hpp]: Yes!

**Comment by dominiklohmann** [comment]
> Mind summarizing why this is problematic?

Ultimately we want the generated FlatBuffers code to be a private dependency of libvast, as it's an implementation detail. We should always develop with the mindset to keep dependencies private where possible.

**Pull request closed without merging**